api/insight/apiroutes.go

```diff
@@ -219,28 +253,96 @@ func (c *insightApiContext) broadcastTransactionRaw(w http.ResponseWriter, r *ht
+	var addresses []string
+	if strings.Contains(address, ",") {
+		addresses = strings.Split(address, ",")
+	} else {
+		addresses = []string{address}
+	}
 
-	var txnOutputs []apitypes.AddressTxnOutput
+	// Initialize Output Structure
+	txnOutputs := make([]apitypes.AddressTxnOutput, 0)
 
 	for _, address := range addresses {
-		if address == "" {
-			http.Error(w, http.StatusText(422), 422)
-			return
+
+		addressOuts, _, _ := c.MemPool.UnconfirmedTxnsForAddress(address)
+
+		confirmedTxnOutputs := c.BlockData.ChainDB.GetAddressUTXO(address)
+
+		// Add mempool tx to the utxo set
+	OUTER:
+		for _, f := range addressOuts.Outpoints {
+			fundingTx, ok := addressOuts.TxnsStore[f.Hash]
+			if !ok {
+				apiLog.Errorf("An outpoint's transaction is not available in TxnStore.")
+				continue
+			}
+			if fundingTx.Confirmed() {
+				apiLog.Errorf("An outpoint's transaction is unexpectedly confirmed.")
+				continue
+			}
+			// TODO: Confirmed() not always return true for txs that have
+			// already been confirmed in a block.  The mempool cache update
+			// process should correctly update these.  Until we sort out why we
+			// need to do one more search on utxo and do not add if this is
+			// already in the list as a confirmed tx.
+			for _, utxo := range confirmedTxnOutputs {
+				if utxo.TxnID == f.Hash.String() && utxo.Vout == f.Index {
+					continue OUTER
+				}
+			}
+
+			txnOutput := apitypes.AddressTxnOutput{}
+			txnOutput.Address = address
+			txnOutput.TxnID = fundingTx.Hash().String()
+			txnOutput.Vout = f.Index
+			txnOutput.ScriptPubKey = hex.EncodeToString(fundingTx.Tx.TxOut[f.Index].PkScript)
+			txnOutput.Amount = dcrutil.Amount(fundingTx.Tx.TxOut[f.Index].Value).ToCoin()
+			txnOutput.Satoshis = fundingTx.Tx.TxOut[f.Index].Value
+			txnOutput.Confirmations = 0
+			txnOutput.BlockTime = fundingTx.MemPoolTime
+
+			txnOutputs = append(txnOutputs, txnOutput)
+		}
+
+		txnOutputs = append(txnOutputs, confirmedTxnOutputs...)
+
+		// Search for items in mempool to remove from the utxo set
+		for _, f := range addressOuts.PrevOuts {
+			spendingTx, ok := addressOuts.TxnsStore[f.TxSpending]
+			if !ok {
+				apiLog.Errorf("An outpoint's transaction is not available in TxnStore.")
+				continue
+			}
+			if spendingTx.Confirmed() {
+				apiLog.Errorf("An outpoint's transaction is unexpectedly confirmed.")
+				continue
+			}
+			for g, utxo := range txnOutputs {
+				if utxo.TxnID == f.PreviousOutpoint.Hash.String() && utxo.Vout == f.PreviousOutpoint.Index {
+					// Found a utxo that is unconfirmed spent.  Remove from slice
+					txnOutputs = append(txnOutputs[:g], txnOutputs[g+1:]...)
+				}
+			}
 		}
-		utxo := c.BlockData.ChainDB.GetAddressUTXO(address)
-		txnOutputs = append(txnOutputs, utxo...)
+		// Final sort by timestamp desc if unconfirmed and confirmations if
+		// confirmed
+		sort.Slice(txnOutputs, func(i, j int) bool {
+			if txnOutputs[i].Confirmations == 0 && txnOutputs[j].Confirmations == 0 {
```

Comment:
Could even have this be `txnOutputs[i].Confirmations == txnOutputs[j].Confirmations`

Reply:
Maybe just we sort by block_time desc.